envoy/common/matchers.h

```diff
@@ -16,10 +18,19 @@ class StringMatcher {
 public:
   virtual ~StringMatcher() = default;
 
+  struct Context {
+    OptRef<const StreamInfo::StreamInfo> stream_info_;
+  };
+
   /**
    * Return whether a passed string value matches.
    */
   virtual bool match(const absl::string_view value) const PURE;
+
+  /**
+   * Return whether a passed string value matches with context
```

Comment:
```suggestion
   * Return whether a passed string value matches with context.
   * Because most implementations don't use the context, provides a
   * default implementation.
```